### PR TITLE
adding extended history content

### PR DESCRIPTION
**Pull request by heckj**
@bossidy @benbp @christineday 

Here's some additional content that I wrote for internal folks on the background and "why we made" RackHD - thought it would be valuable as an extension onto the introduction text. No promises on my "grammar" as usual, so looking for a real editorial read-through & suggestions to fix this up before I merge it in.

If you'd prefer to "merge and fix after", I'm cool with that - just let me know.

-joe

### CODE REVIEW

File: docs/introduction.rst [hunk 1]
```diff
@@ -80,6 +80,109 @@ services providing classical PXE booting makes it possible to architect a number
...
+ * interrogate the hardware, see if we’re at the right version of firmware
+ * if not, flash the firmware to the version we want
+ * reboot (mandated by things like BIOS and BMC flashing)
+ * PXE boot again
+ * interrogate the hardware
+ * make sure we’re at the right version of firmware
+ * SCORE!
+
+To achieve this, the existing systems (Cobbler, Razor, etc) needed another
+evel of orchestration - resetting what we PXE boot, interacting with data from
```
Comment (benbp) [inline docs/introduction.rst]: "evel" vs. "level"

File: docs/introduction.rst [hunk 2]
```diff
@@ -80,6 +80,109 @@ services providing classical PXE booting makes it possible to architect a number
...
+he machine in question, and making some logical decisions. This sequence of
```
Comment (benbp) [inline docs/introduction.rst]: "he" vs. "the"

File: docs/introduction.rst [hunk 3]
```diff
@@ -80,6 +80,109 @@ services providing classical PXE booting makes it possible to architect a number
...
+the lowest level of automation supporting interrogating hardware, setting a
+“personality” onto it (in the form of an OS), providing consistent REST based
+API’s for controlling that hardware - agnostic of hardware vendor, and for
+using the pollers to capture telemetry - create “live data feeds” that can be
+provided via a pub/sub style interface.
+
+As we went through use cases and expanded features, we made the capability for
+the workflow engine to react to what it discovered - what we call “SKU” support,
+dynamic rendering of templates for OS installs, and passing of variables and
+data from the APIs that invoke workflow through to the configuration files that
```
Comment (benbp) [inline docs/introduction.rst]: "that invoke workflow*s*"

**Comment by benbp** [comment]
A couple minor typo notes. Looks good, though a general note to be aware of: the vernacular reads more like a Joe e-mail than official docs (although perhaps this is just because I know your voice, idk).

**Comment by heckj** [comment]
@benbp typos fixed, thank you. And yeah, it's totally in my colloquial voice. Ben, suggestions for how you'd change or update this to make it less "joe-specific"?

**Comment by benbp** [comment]
I think there are four main things for me that contribute to it sounding colloquial:

1. references to "we" and "us" and/or docs told from the perspective of a single person.
2. Opinions offered (e.g. "while not without it’s quirks") vs. declarative statements ("there are limitations to X")
3. Similar to 1, references to "they" (e.g. "**they** have a microkernel"). Reinforces the language as coming from a person relating to other people rather than being in the abstract (the Hanlon project includes a microkernel).
4. Possession of things (also e.g. "they **have** a microkernel"). Makes it sound like we're analyzing a competitor internally rather than presenting a comparison argument for public reading.

Basically, any language that contributes to sounding like the docs coming from the voice of a single person rather than not. Maybe one litmus test that could be used is: if someone else has to edit these docs, will that make sense? So if someone new goes in and updates this specific doc, they would be editing language that says "we did this" when they weren't part of the "we" being referred to.

**Comment by heckj** [comment]
cool good markers, I'll take a pass through this and try and rewrite it with that in mind, getting rid of the possessives and personal

**Comment by bossidy** [comment]
I'm going to go ahead and merge. Then I'll work on the language to make it sound less colloquial

**Comment by bossidy** [comment]
Gee. I don't think moving the Glossary to the "How It Works" section is a good idea. It distracts from the purpose of the section. Ideally, a "glossary" section should be separate, because it can serve the entire documentation. 

My 2 cents................

**Comment by heckj** [comment]
It was the only way I found to reference the content with the links as I did. I was aiming to make that very top section a general "this is what PxE is and how it works" to provide a background for all the rest of the detail, which is only relevant if you grok pxe

**Comment by heckj** [comment]
By itself the glossary was fairly useless, as you pointed out, so maybe it's more sensible to make it a more direct narrative flow describing how pxe works in that first paragraph, and kill the concept of the glossary section all together

**Comment by bossidy** [comment]
I think that would work, especially if we use links to wikipedia for general reference. I'll work on it.

BTW, are we sticking to rst or moving to markdown?

**Comment by heckj** [comment]
I haven't dig into the example you provided earlier, so for now I've just been sticking with restructured text since we're already down that road
